### ISSUE TITLE
Reduce effort with custom keywords in shared stories.

### ISSUE THREAD

**Issue by staffehn** [fabd/kanji-koohii#139]
Hey. As a non-English user I’m using custom keywords a lot, hence I noticed two situations where I have to reformat stories during the sharing process when I wouldn’t really expect it.

First situation is when sharing a story. An occurrence of a custom keyword (*not* surrounded by #’s) that gets highlighted in bold for me won’t be bold in the shared story anymore, since there only the original keyword get’s highlighted.

Second situation is when copying someone else’s shared story. Then an occurrence of the default keyword (that’s not already surrounded by #’s by the author) will not be bold in my copy, unless I manually insert some #’s myself. Yes I do quite often copy an English story without translating it (that’d cost more time) as the story is only used in case I forget it, anyways, but I still need the English (original) keyword there be highlighted then.

I’d suggest to fix those by changing two things.

Firstly, you should have the authors custom keyword available for a shared story rendering, so that you can use that information to highlight it correctly (or maybe the custom keyword is available in your flashcards classes anyways any you’d only have to change the code that displays it to use that custom keyword).

Secondly, when copying another custom story, if the own keyword is different from the story authors keyword (so either I have a custom keyword and they use default, I have default and they use custom, or two different custom keywords), then the website should add #’s around the original keyword occurrences such that they’re still going to be highlighted in my copy. That’s going to be a little bit more complex to archive for the reason that you shouldn’t add any extra #’s if there are already some #’s present, especially if those already present #’s are not directly enclosing the word but also contain some spaces for example (or even more).
Also, then, if you like, you could also call that function every time I change a custom keyword, so that the old keyword is still staying highlighted. (Pretty unimportant case, but when there’s code to reuse, why not :D)

**Comment by fabd** [comment]
Hi staffehn

This is a won't fix.

For #1 the only solution that makes sense in terms of performance is that, in retrospect, the formatting should have always been included when sharing a story. But it isn't the case. The shared list uses a "on the fly" formatting based on default keyword. To request each user's personalized keyword would be a massive performance impact. Which is why, again in retrospect, it would have been better to make all the formatting *explicit* when sharing a story. -- So even if I fixed this now, you wouldn't benefit from it since most upvoted stories are years old and wouldn't be updated unless I run some batch script... a big hassle for very little benefit, and a time spent that can not be justified seeing as majority of users don't systematically update most of their keywords. (the real issue here, being that Koohii has never had multi lingual support, AND wouldn't get the [ french / spanish / etc ] author's permission anyway).

For #2, sure it could be done, and not very complex, but time I can not justify either.

In conclusion these are in the category of "annoyances". which at this point of the site's life, are not justifiable for me to spend time fixing.

We could also make a case that any custom keywords should be explicitly formatted by the user as we have no idea what they are, and sometimes they are parttially or not highlighted due to inflection / conjugation (ie. verbs).
